developing/vcs/README.md

```diff
@@ -2,7 +2,225 @@
+changes.
+
+```
+git fetch origin
+git rebase origin/master
+```
+
+Resolve any conflicts manually. If the conflicted file was deleted on the master
+branch, remove it from your branch:
+
+```
+git rm <conflicted_file>
+```
+
+Otherwise, add it:
+
+```
+git add <conflicted_file>
+```
+
+Continue the rebase:
+
+```
+git rebase --continue
+```
+
+If you've created more than one commit, use
+`[git's interactive rebase](https://help.github.com/en/articles/about-git-rebase)`
+to squash them into cohesive commits with good messages:
+
+```
+git rebase -i origin/master
+```
+
+For each meaningful commit, write a
+[good commit message](https://chris.beams.io/posts/git-commit/) following these
+seven rules:
+
+1. Separate subject from body with a blank line
+2. Limit the subject line to 50 characters
+3. Capitalize the subject line
+4. Do not end the subject line with a period
+5. Use the imperative mood in the subject line
+6. Wrap the body at 72 characters
+7. Use the body to explain what and why vs. how
+
+Example:
+
+```
+Replace chromedriver-helper with webdrivers
+
+`chromedriver-helper` says in bold on their README that they are no
+longer maintaining it and that people should use `webdrivers` instead.
+
+Additionally, `webdrivers` has a very useful feature that will
+automatically use the right version of `chromedriver` based on the
+version of Chrome that is installed on the machine. This was not
+possible with `chromedriver-helper`, which made it harder to deal with
+versioning locally and in Circle CI.
+
+Resolves #123
+```
+
+Push your branch:
+
+```
+git push origin <branch-name>
+```
+
+Submit a GitHub pull request.
+
+Ask for a [code review](../code-reviews/README.md) in the project's Slack
+channel, and/or via the GitHub site.
+
+Once you push to GitHub, only make changes via new commits, as opposed to
+amending or rebasing, to make it easier for reviewers to see what has changed
+since they last provided feedback. If you need to make a minor change to an
+earlier commit, take advantage of the `--fixup` flag, coupled with the
+`--autosquash` flag for interactive rebase, which you should enable globally:
+
+```
+git config --global rebase.autosquash true
+```
+
+```
+git commit --fixup <first-few-chars-of-earlier-commit-SHA>
+```
+
+You can also search for the commit by a string in its message:
+
+```
+git commit --fixup :/unique-string-in-recent-commit-message
+```
+
+thoughtbot has a great blog post about
+[auto-squashing git commits](https://thoughtbot.com/blog/autosquashing-git-commits).
+
+Once the PR has been approved, rebase against master again to pick up any
+changes:
+
+```
+git fetch origin
+git rebase origin/master
+```
+
+Fix conflicts as before if necessary, and continue the rebase. Then squash the
+commits you made during code review into meaningful ones:
+
+```
+git rebase -i origin/master
+```
+
+Force push the branch back up to GitHub:
+
+```
+git push --force-with-lease origin <branch-name>
```

Comment:
This is the safer option, but you can still trick --force-with-lease into overwriting stuff especially if you are using `git fetch` vs `git pull` to update your branch. Here's a helpful post that we can link here for more details https://blog.developer.atlassian.com/force-with-lease/ though I still think this is probably better as part of an advanced usage section.